crates/typst-pdf/src/page.rs

```diff
@@ -150,11 +141,40 @@ fn write_global_resources(ctx: &mut PdfContext) -> Ref {
     // Write all of the functions used by the document.
     ctx.colors.write_functions(&mut ctx.pdf);
 
-    resource_ref
+    // Also write the resources for Type3 fonts, that only contains images,
+    // color spaces and regular fonts (COLR glyphs depend on them).
+    let mut resources =
+        ctx.pdf.indirect(ctx.type3_font_resources_ref).start::<Resources>();
+
+    ctx.colors
+        .write_color_spaces(resources.color_spaces(), &mut ctx.alloc);
+
+    let mut images = resources.x_objects();
```

Comment:
By storing the XObjects, Fonts, and ExtGStates in indirect objects, we could just refer to them here instead of duplicating everything. That will require dropping down the untyped pdf-writer API. Unfortunately, pdf-writer forces direct objects in many places. :/ We need some more flexible solution medium-term.

I think we can also skip this section entirely if there are no Type3 fonts.